source/extensions/transport_sockets/tls/cert_validator/cert_validator.h

```diff
@@ -46,7 +46,10 @@ struct ValidationResults {
 class CertValidator {
 public:
   // Wraps cert validation parameters added from time to time.
-  struct ExtraValidationContext {};
+  struct ExtraValidationContext {
+    // Local socket address of caller that connects to remote validation server (optional field).
+    Envoy::Network::Address::InstanceConstSharedPtr local_address;
```

Comment:
Would it make sense to store the `connectionInfoProvider()` here instead? Then if another user needed the remote address it would be available (as well as a few other things).

Reply:
@danzh2010 , will quic stack have this info and will it work for quick stack?

@ggreenway  Context: local address was added like common divisor between quic and tcp stack. That is reason we only add what we need.

Reply:
Oh, that makes sense. I'm fine leaving it as just local_address then.